fvm/evm/emulator/emulator.go

```diff
@@ -234,7 +234,17 @@ func (bl *BlockView) DryRunTransaction(
 	msg.SkipAccountChecks = true
 
 	// return without commiting the state
-	return proc.run(msg, tx.Hash(), 0, tx.Type())
+	txResult, err := proc.run(msg, tx.Hash(), 0, tx.Type())
+	if txResult.VMError == nil && txResult.ValidationError == nil {
```

Comment:
let's add `Successful() bool` to the result that internally does `return !Failed() && !Invalid()`

Reply:
Good point :+1: 
Added in https://github.com/onflow/flow-go/pull/6059/commits/b6fc9e1ea349062b7fbbb9d575421f9d213518e2